includes/widgets/class-wc-widget-layered-nav.php

```diff
@@ -365,9 +365,10 @@ protected function get_filtered_term_product_counts( $term_ids, $taxonomy, $quer
 		$main_tax_query_sql = $this->convert_tax_query_to_sql( $main_tax_query );
 		$term_ids_sql       = '(' . implode( ',', array_map( 'absint', $term_ids ) ) . ')';
 
-		// Generate query.
+		// Generate the first part of the query.
+		// This one will return non-variable products and variable products with concrete values for the attributes.
 		$query           = array();
-		$query['select'] = "SELECT COUNT( DISTINCT {$wpdb->posts}.ID ) as term_count, terms.term_id as term_count_id";
+		$query['select'] = "SELECT {$wpdb->posts}.post_parent as product_id, terms.term_id as term_count_id";
```

Comment:
are we sure about this change? `post_parent` is set to 0 for normal products, i think this is applicable only for variations.